client/src/pages/result/Result.js

```diff
@@ -3,9 +3,12 @@ import rawData from "../../data/kujiHelplineSheet.json";
 import Table from "react-bootstrap/Table";
 import  "./Result.css";
 import Form from "react-bootstrap/Form";
+import { useSearchParams } from "react-router-dom";
+
 export default function Result() {
-  const [service,setService]=useState("Legal Aid");
-  const [location,setLocation]=useState("North");
+  let [searchParams, setSearchParams] = useSearchParams();
+  const [service,setService]=useState(searchParams.get("service"));
```

Comment:
This is a good opportunity to explore implementing a custom hook.
See the official docs on [custom hooks](https://reactjs.org/docs/hooks-custom.html).

The pattern I see here is that we want to keep the state and seach params synchronised, so why don't we write our own special version of `useState` called `useSearchParamState` which combines `useState` and `useSearchParams` ?

Reply:
That is a good idea but need more for me to learn more about it. Can we do this in the next iteration?